src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java

```diff
@@ -76,6 +76,45 @@ public EventDTO save(EventDTO eventDTO) {
             .map(eventMapper::toDto);
     }
 
+    /**
+     * Get all the upcoming events.
+     *
+     * @return the list of entity.
+     */
+    @Override
+    public Page<EventDTO> findAllUpcomingEvents(Pageable pageable) {
+        log.debug("Request to get all Upcoming Events");
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.OPEN);
+            add(EventStatus.POSTPONED);
+        }};
+
+        Instant from = Instant.now();
+
+        return eventRepository.findEventsByStartDateAfterAndStatusIn(from, eventStatuses, pageable)
+            .map(eventMapper::toDto);
+    }
+
+    /**
+     * Get all the past events.
+     *
+     * @return the list of entity.
+     */
+    @Override
+    public Page<EventDTO> findAllPastEvents(Pageable pageable) {
+        log.debug("Request to get all Past Events");
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.CANCELLED);
```

Comment:
If CANCELLED why still need to show in Previous Event?

Reply:
So means if the event is cancelled, the admin can't visit the event again?
As my concerned is let say when an admin accidently changed the status to cancelled, the admin should can look for the event and change back the status.

Reply:
we can have a discussion on this on Friday
so far is ok I think